kubectl-rook-ceph.sh

```diff
@@ -16,33 +16,269 @@
+        break
+        ;;
+    esac
+    is_flag "$VAL" && fail_error "Flag '$FLAG' value '$VAL' looks like another flag"
+    # run the command with the current value, which may be empty
+    if ! $set_value_function "$FLAG" "$VAL"; then
+      # the flag needs a value, so grab the next arg to use as the value
+      VAL="$1" || fail_error "Could not get value for flag '$FLAG'"
+      shift
+      # fail if the next arg looks like a flag and not a value
+      is_flag "$VAL" && fail_error "Flag '$FLAG' value '$VAL' looks like another flag"
+      # fail because the flag needs a value and value given is empty, e.g., --namespace ''
+      val_exists "$VAL" || fail_error "Flag '$FLAG' does not specify a value"
+      # run the command again with the next arg as its value
+      if ! $set_value_function "$FLAG" "$VAL"; then
+        fail_error "Flag '$FLAG' must have a value" # probably won't reach this, but just in case
+      fi
     fi
+  done
+}
+
+# call this at the end of a command tree when there should be no more inputs past a given point.
+# Usage: end_of_command_parsing "$@" # where "$@" contains the remaining args
+function end_of_command_parsing () {
+  if [[ "$#" -gt 0 ]]; then
+    fail_error "Extraneous arguments at end of input: $*"
+  fi
+}
+
+####################################################################################################
+# 'kubectl rook-ceph help' command
+####################################################################################################
+
+function run_help_command () {
+  end_of_command_parsing "$@" # end of command tree
+  print_usage
+}
+
+####################################################################################################
+# 'kubectl rook-ceph ceph ...' command
+####################################################################################################
+
+function run_ceph_command () {
+  # do not call end_of_command_parsing here because all remaining input is passed directly to 'ceph'
+  kubectl --namespace "$NAMESPACE" exec deploy/rook-ceph-operator -- /bin/bash -c " ceph ${*} --conf=/var/lib/rook/rook-ceph/rook-ceph.config"
+}
+
+####################################################################################################
+# 'kubectl rook-ceph operator ...' commands
+####################################################################################################
+
+function run_operator_command () {
+  [[ -z "${1:-""}" ]] && fail_error "Missing 'operator' subcommand"
+  subcommand="$1"
+  shift # remove the subcommand from the front of the arg list
+  case "$subcommand" in
+    restart)
+      run_operator_restart_command "$@"
+      ;;
+    *)
+      fail_error "'operator' subcommand '$subcommand' does not exist"
+      ;;
+  esac
+}
+
+function run_operator_restart_command () {
+  end_of_command_parsing "$@" # end of command tree
+  kubectl --namespace "$NAMESPACE" rollout restart deploy/rook-ceph-operator
```

Comment:
for now, I'm removing it as when operator restart is run it will show relevant log by kubectl so not required IMO for now. something like `deployment.apps/rook-ceph-operator restarted`